tools/devtool

```diff
@@ -1432,6 +1448,70 @@ cmd_install() {
     install -m 755 "$CARGO_SECCOMPILER_TARGET_DIR/$target/$profile/seccompiler-bin" "$install_path"
 }
 
+cmd_build_kernel() {
+    nprocs=2
+
+    # Parse any command line args.
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            "-h"|"--help")      { cmd_help; exit 1; } ;;
+            "-c"|"--config")
+                shift
+                local KERNEL_CFG="$1"
+                ;;
+            "-n"|"--nproc")
+                shift
+                nprocs="$1"
+                ;;
+            *)
+                die "Unknown argument: $1. Please use --help for help."
+            ;;
+        esac
+        shift
+    done
+
+    # It is mandatory to provide a valid config file for building the kernel.
+    [ -z "$KERNEL_CFG" ] && die "You need to provide the path to a valid kernel config!"
+
+    arch=$(uname -m)
+    if [ "$arch" = "x86_64" ]; then
+        target="vmlinux"
+        cfg_pat="x86"
+        format="elf"
+    elif [ "$arch" = "aarch64" ]; then
+        target="Image"
+        cfg_pat="arm64"
+        format="pe"
+    else
+        die "Unsupported architecture!"
+    fi
+
+    kernel_dir_host="$FC_BUILD_DIR"/kernel
+    kernel_dir_ctr="$CTR_FC_BUILD_DIR"/kernel
+    create_dir "$kernel_dir_host"
+
+    # Extract the kernel version from the config file provided as parameter.
+    KERNEL_VERSION=$(cat "$KERNEL_CFG" | grep -Po "^# Linux\/$cfg_pat (([0-9]+.){2}[0-9]+)" | cut -d ' ' -f 3)
+    validate_kernel_version "$KERNEL_VERSION"
```

Comment:
yes, `yes "" | make oldconfig` would generate a default config. Also, in case the provided config is fully compatible, it wouldn't alter it. In case it isn't fully compatible it would alter it, and we could check that and issue a warning. So we would reach a similar outcome as above. Seems more reliable to me then validating a comment that can be wrong or modified by the user. Also we would avoid regexes and pattern matching which I personally find hard to follow.